File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0;
        int max = nums[0];
        for (int i=0; i < nums.length; i++) {
            if (count == 0) {
                max = nums[i];
            }

            if (nums[i] == max) {
                count++;
            } else {
                count--;
            }
        }

        return max;
        // int verify = 0;
        // for (int i=0; i < nums.length; i++) {
        //     if (nums[i] == max) verify++;
        // }

        // if (verify > nums.length / 2) return max;
        // else return -1;
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int maxi = -1;
        int count = 0;

        for (int i=0; i < n; i++) {
            if (count == 0) {
                maxi = nums[i];
                count = 1;
            } else {
                if (nums[i] == maxi) {
                    count++;
                } else {
                    count--;
                }
            }
        }

        // if (count == 0) {
        //     for (int i=0; i < n; i++) {
        //         if (nums[i] == maxi) {
        //             count++;
        //         }
        //     }
        //     if (count > (n / 2)) return maxi
        // }
        return maxi;
    }
};